.github/workflows/velox_weekly.yml

```diff
@@ -63,8 +62,12 @@ jobs:
           export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk && \
           export PATH=$JAVA_HOME/bin:$PATH
           
-          # The upstream main branch is cloned. Please temporarily switch to the dev branch for PR validation.
+          # action/checkout does not work centos7 anymore, so we clone the branch instead.
           git clone -b main --depth=1 https://github.com/apache/incubator-gluten.git && cd incubator-gluten/
+          # The branch of a pull request is a special remote reference, we need to clone the repo first, then fetch it with pr ref.
+          if [ ${{ github.event_name }} = "pull_request" ]; then
+            git fetch origin ${{ github.ref }}:pr_branch && git checkout pr_branch
```

Comment:
Here chechout the pull_request pr, so we don't need to manually switch branch to verify the result.